src/scrap.py

```diff
@@ -0,0 +1,262 @@
+    # Inserts opportunities if and only if they do not already exist
+
+    connection = sqlite3.connect("jobs.db")
+    cursor = connection.cursor()
+
+    for job in job_data:
+        cursor.execute(
+            "SELECT COUNT(*) FROM jobs WHERE _company = ?, AND _title = ?, AND _location = ?",
+            (job["_company"], job["_title"], job["_location"]),
+        )
+        counter = cursor.fetchone()[0]
+
+        # https://pynative.com/python-cursor-fetchall-fetchmany-fetchone-to-read-rows-from-table/
+
+        if counter == 0:  # The Job Exists if the counter is 0
+            cursor.execute(
+                "INSERT INTO jobs VALUES (?, ?, ?, ?, ?)",
+                (job["_company"], job["_title"], job["_location"], job["_link"], False),
+            )
+        else:
+            continue
+
+    connection.commit()
+    connection.close()
+
+
+# -------------------FOR REDDIT----------------------------
+
+
+# TO-DO: Fix this up to add the company and the location
+def reddit_scrape() -> List[object]:
+    # Web Scrapping Tools
+    software_engineer_url = os.getenv("SOFTWARE_ENGINEER_URL")
+
+    response = requests.get(software_engineer_url)
+
+    content = response.text
+
+    parse_content = BeautifulSoup(content, "html.parser")
+
+    div_post = parse_content.find_all("div", class_="_1poyrkZ7g36PawDueRza-J")
+
+    reddit_jobs = []
+
+    for element in div_post:
+        header_title = element.find("h3", class_="_eYtD2XCVieq6emjKBH3m")
+        link_job = element.find("a", class_="_3t5uN8xUmg0TOwRCOGQEcU")
+
+        job = {}
+
+        if link_job is not None:
+            job["_title"] = header_title.text
+            job["_link"] = link_job.get("href")
+
+            reddit_jobs.append(job)
+
+    return reddit_jobs
+
+
+# -------------------FOR RAPIDAPI----------------------------
+
+
+def rapid_response() -> List[object]:
+    url = "https://indeed12.p.rapidapi.com/jobs/search?locality=us&query=software+engineer&location=Fullerton&formatted_relative_time=today&page_id=1"
+
+    rapid_api_key = os.getenv("RAPID_API_KEY")
+    rapid_api_host = os.getenv("RAPID_API_HOST")
+
+    headers = {
+        "X-RapidAPI-Key": rapid_api_key,
+        "X-RapidAPI-Host": rapid_api_host,
+    }
+
+    rapid_jobs = []
+    response = requests.get(url, headers=headers).json()
+
+    for elem in response["hits"]:
+        job = {}
+        job["_company"] = elem["company_name"]
+        job["_title"] = elem["title"]
+        job["_location"] = elem["location"]
+        job["_link"] = f'https://www.indeed.com/viewjob?jk={elem["id"]}&locality=us'
+
+        rapid_jobs.append(job)
+
+    return rapid_jobs
+
+
+# -------------------FOR INDEED API----------------------------
+
+
+# -------------------FOR LINKEDIN----------------------------
+
+
+def indeed_response() -> List[object]:
+    url = os.getenv("LINKEDIN_URL")
+
+    response = requests.get(url)
+
+    content = response.text
+
+    parse_content = BeautifulSoup(content, "html.parser")
```

Comment:
how come not just use response.text here instead of content when only used once

Reply:
it just looks prettier
😄